doc/source/design.rst

```diff
@@ -98,6 +90,15 @@ supports only multiplication and addition over a finite field, by
 applying a generic set intersection protocol from the standard library
 that is itself only based on a basic arithmetic factory.
 
+We have in FRESCO taken this concept a bit further. We also introduced the
+concept of *builders*. A builder resembles the ``BigInteger`` class in style and
+use. Their methods work on either binary or arithmetic types and produces the
+result of a given function as the return type. This makes it possible to write
+really concise expressions which are easy to read. Examples can be found in any
+test within the framework or here in the documentation in the
+:ref:`Quickstart<Quickstart>` section. Such a builder is given to the
+application developer when implementing the FRESCO ``Application`` interface.
+
```

Comment:
Maybe add something about the fact that there are two types implemented you can use, specialize or even build your own